docs/user-guide/horizontal-pod-autoscaling/walkthrough.md

```diff
@@ -120,6 +127,145 @@ Here CPU utilization dropped to 0, and so HPA autoscaled the number of replicas
+by making use of the `autoscaling/v2alpha1` API version.
+
+First, get the YAML of your HorizontalPodAutoscaler in the `autoscaling/v2alpha1` form:
+
+```shell
+$ kubectl get hpa.autoscaling.v2alpha1 -o yaml > /tmp/hpa-v2.yaml
+```
+
+Open the `/tmp/hpa-v2.yaml` file in an editor, and you should see YAML which looks like this:
+
+```yaml
+apiVersion: autoscaling/v2alpha1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: php-apache
+  namespace: default
+spec:
+  scaleTargetRef:
+    apiVersion: extensions/v1beta1
+    kind: Deployment
+    name: php-apache
+  minReplicas: 1
+  maxReplicas: 10
+  metrics:
+  - type: Resource
+    resource:
+      name: cpu
+      targetAverageUtilization: 50
+status:
+  observedGeneration: 1
+  lastScaleTime: <some-time>
+  currentReplicas: 1
+  desiredReplicas: 1
+  currentMetrics:
+  - type: Resource
+    resource:
+      name: cpu
+      currentAverageUtilization: 0
+      currentAverageValue: 0
+```
+
+Notice that the `targetCPUUtilizationPercentage` field has been replaced with an array called `metrics`.
+The CPU utilization metric is a *resource metric*, since it is represented as a percentage of a resource
+specified on pod containers.  Notice that you can specify other resource metrics besides CPU.  By default,
+the only other supported resource metric is memory.  These resources do not change names from cluster
+to cluster, and should always be available, as long as Heapster is deployed.
+
+You can also specify resource metrics in terms of direct values, instead of as percentages of the
+requested value.  To do so, use the `targetAverageValue` field insted of the `targetAverageUtilization`
+field.
+
+There are two other types of metrics, both of which are considered *custom metrics*.  These metrics
+may have names which are cluster specific, and require a more advanced cluster monitoring setup.
+
+The first of these alternative metric types is *pod metrics*.  These metrics describe pods, and
+are averaged together across pods and compared with a target value to determine the replica count.
+They work much like resource metrics, except that they *only* have the `targetAverageValue` field.
+
+Pod metrics are specified using a metric block like this:
+```yaml
+type: Pods
+pods:
+  metricName: packets-per-second
+  targetAverageValue: 1k
+```
+
+The second alternative metric type is *object metrics*.  These metrics describe a different
+object in the same namespace, instead of describing pods.  Note that the metrics are not
+fetched from the object -- they simply describe it.  Object metrics do not involve averaging,
+and look like this:
+
+```yaml
+type: Object
+object:
+  metricName: requests-per-second
+  target:
+    apiVersion: extensions/v1beta1
+    kind: Ingress
+    name: main-route
+  targetValue: 2k
+```
+
+You can provide multiple such metric blocks to have the HorizontalPodAutoscaler consider multiple metrics.
```

Comment:
This sentence is a bit awkward, but I am not sure how to rewrite it. I am not sure what you are trying to emphasize. 
"If you provide multiple metric blocks, the HorizontalPodAutoscaler will consider each block as a separate metric." ??
"The HorizontalPodAutoscaler can consider multiple metrics if the metrics are in separate metric blocks." ??